libsplinter/src/admin/rest_api/actix/ws_register_type.rs

```diff
@@ -61,34 +63,68 @@ pub fn make_application_handler_registration_route<A: AdminCommands + Clone + 's
+            #[cfg(not(feature = "admin-service-event-store"))]
+            let initial_events = {
+                let (skip, last_seen_timestamp) = query
+                    .remove("last")
+                    .map(|since_millis| {
+                        // Since this is the last seen event, we will skip it in our since
+                        // query
+                        debug!("Catching up on events since {}", since_millis);
+                        (
+                            1usize,
+                            time::SystemTime::UNIX_EPOCH
+                                + time::Duration::from_millis(since_millis),
+                        )
+                    })
+                    .unwrap_or((0, time::SystemTime::UNIX_EPOCH));
+
+                match admin_commands
+                    .get_events_since(&last_seen_timestamp, &circuit_management_type)
+                {
+                    Ok(events) => events.map(JsonAdminEvent::from).skip(skip),
+                    Err(err) => {
+                        error!(
+                            "Unable to load initial set of admin events for {}: {}",
+                            &circuit_management_type, err
+                        );
+                        return Box::new(
+                            HttpResponse::InternalServerError().finish().into_future(),
+                        );
+                    }
+                }
+            };
+
+            #[cfg(feature = "admin-service-event-store")]
+            let initial_events = {
+                let (skip, last_seen_event_id) = query
+                    .remove("last")
+                    .map(|since_evt_id| {
+                        // Since this is the last seen event, we will skip it in our since
+                        // query
+                        let id: i64 = since_evt_id.try_into().unwrap_or(0);
+                        debug!("Catching up on events since {}", id);
+                        (1usize, id)
+                    })
+                    .unwrap_or((0, 0));
+
+                match admin_commands.get_events_since(&last_seen_event_id, &circuit_management_type)
+                {
+                    Ok(events) => events.map(JsonAdminEvent::from).skip(skip),
+                    Err(err) => {
+                        error!(
+                            "Unable to load initial set of admin events for {}: {}",
+                            &circuit_management_type, err
+                        );
+                        return Box::new(
+                            HttpResponse::InternalServerError().finish().into_future(),
+                        );
+                    }
                 }
             };
 
             let request = Request::from((request, payload));
-            match new_websocket_event_sender(request, Box::new(initial_events.skip(skip))) {
+            match new_websocket_event_sender(request, Box::new(initial_events)) {
```

Comment:
Why was this skip removed?

Reply:
The skip wasn't removed, I just moved it up to use `skip` where `initial_events` is being returned.